Sources/_StringProcessing/Regex/Core.swift

```diff
@@ -132,3 +135,10 @@ extension Regex {
     self.program = Program(tree: .init(node))
   }
 }
+
+@available(SwiftStdlib 5.7, *)
+extension Regex {
+  internal mutating func _setCompilerOptionsForTesting(_ opts: Compiler.CompileOptions) {
```

Comment:
Hmm, my concern is that this might not interact well with how we cache compiled programs. How would a caller compile and run it both ways?